community/kernel/src/main/java/org/neo4j/kernel/impl/store/StoreVersionTrailerUtil.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.store;
+
+import java.io.IOException;
+
+import org.neo4j.helpers.UTF8;
+import org.neo4j.io.pagecache.PageCursor;
+import org.neo4j.io.pagecache.PagedFile;
+import org.neo4j.kernel.impl.util.Charsets;
+
+public abstract class StoreVersionTrailerUtil
+{
+
+    public static long getTrailerOffset( PagedFile pagedFile, String trailer )
+            throws IOException
+    {
+        int trailerOffset;
+        byte[] encodedTrailer = UTF8.encode( trailer );
+        int maximumNumberOfPagesVersionSpans = getMaximumNumberOfPagesVersionSpans( encodedTrailer.length, pagedFile );
+        long pageId = Math.max( pagedFile.getLastPageId() + 1 - maximumNumberOfPagesVersionSpans, 0 );
+        try ( PageCursor pageCursor = pagedFile.io( pageId, PagedFile.PF_SHARED_LOCK ) )
+        {
+            byte[] allData = getLastBytes( pagedFile, maximumNumberOfPagesVersionSpans, pageCursor );
+            trailerOffset = findTrailerOffset( allData, encodedTrailer );
+        }
+        return trailerOffset + pageId * pagedFile.pageSize();
+    }
+
+    public static void writeTrailer( PagedFile pagedFile, byte[] trailer,
+            long trailerOffset ) throws IOException
+    {
+        long pageId = trailerOffset / pagedFile.pageSize();
+        try ( PageCursor pageCursor = pagedFile.io( pageId, PagedFile.PF_EXCLUSIVE_LOCK ) )
```

Comment:
I am not sure I understand why `pageId` is computed this way. Shouldn't you write the trailer starting from `pagedFile.getLastPageId()`?